app/models/generic_object_definition.rb

```diff
@@ -0,0 +1,12 @@
+class GenericObjectDefinition < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  serialize :properties, Hash
+
+  has_one   :picture, :dependent => :destroy, :as => :resource
+  has_many  :generic_objects
+
+  def defined_attributes
+    @defined_attributes ||= properties[:attributes].stringify_keys
+  end
```

Comment:
a `properties=` setter should probably be changed to clear this cache...or perhaps just don't cache it at all here.